lm_eval/models/huggingface.py

```diff
@@ -497,26 +499,28 @@ def _collate(x):
 
         re_ord = utils.Reorderer([req.args for req in requests], _collate)
 
-        for context, gen_kwargs in tqdm(re_ord.get_reordered()):
+        for context, gen_kwargs in tqdm(
+            re_ord.get_reordered(), disable=(self.rank != 0)
+        ):
             until = None
             if isinstance(gen_kwargs, dict):
-                gen_kwargs = copy.deepcopy(gen_kwargs)  # edge case for repeats > 1
-                if "until" in gen_kwargs.keys():
-                    until = gen_kwargs.pop("until")
+                kwargs = copy.deepcopy(gen_kwargs)  # edge case for repeats > 1
+                if "until" in kwargs.keys():
+                    until = kwargs.pop("until")
                     if isinstance(until, str):
-                        until = [gen_kwargs]
+                        until = [kwargs]
                     elif not isinstance(until, list):
                         raise ValueError(
-                            f"Expected `gen_kwargs['until']` to be of type Union[str,list] but got {until}"
+                            f"Expected `generation_kwargs['until']` to be of type Union[str,list] but got {until}"
```

Comment:
Shouldn't this be `kwargs` instead of `generation_kwargs`? The variable `until` is observed from `kwargs.pop("until")

Reply:
Fixed!